src/main/java/com/es/phoneshop/model/entity/cart/CartItem.java

```diff
@@ -0,0 +1,45 @@
+package com.es.phoneshop.model.entity.cart;
+
+import com.es.phoneshop.model.entity.product.Product;
+
+import java.util.Objects;
+
+public class CartItem {
+    private Product product;
+    private int quantity;
+
+    public CartItem(Product product, int quantity) {
+        this.product = product;
+        this.quantity = quantity;
+    }
+
+    public Product getProduct() {
+        return product;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    @Override
+    public String toString() {
+        return product.getCode() +
+                ": " + quantity;
+    }
+    public void addToQuantity(int quantity) {
+        this.quantity += quantity;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        CartItem cartItem = (CartItem) o;
+        return Objects.equals(product, cartItem.product);
```

Comment:
Your equals doesn't consider quantity. It's better to include essential fields in equals